src/main/java/com/solvd/ecommerce/ResultPage.java

```diff
@@ -0,0 +1,38 @@
+package com.solvd.ecommerce;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+
+import java.util.List;
+
+public class ResultPage {
+
+    private final WebDriver webDriver;
+
+    @FindBy(css = ".item-type-card")
+    private List<WebElement> resultTableElementList;
```

Comment:
The css locator for your title can be **.item-type-card__item .item-type-card__title**, you need to other details of cards by text on the cards, using for example xpath

Reply:
From what I have found I cannot make dynamic changes to fields with FindBy annotation, thus at least one .findElement method seems to be inevitable. Have I got it right?

Reply:
Yes, you cannot change element with annotation findBy, but in selenium it's possible to generate dynamic elements, for example using this structure **By.xpath(String.format("your_xpath_here", "your_string"))**
The same actual for cssSelector, className and others

In carina framework it's easier, because we use ExtendedWebElement, and you can use format for every element with annotation FindBy: someExtendedWebElement.format("some_string"), but it'll be in the future :)

Reply:
Now I've got it, thank you! 
The structure is really convenient to use with pure Selenium, but I'm afraid that in my particular case it makes the code way too complicated to implement. A lot of items contain either double quotes or an apostrophe in their name, which I couldn't escape with a " \" " or " \' ", so I had to write some extra code to create a valid xpath that would let the test run:
`public void printItemData() {
        resultNameList.forEach(el -> {
            String brandXPath = null;
            String itemName = el.getText();
            if (itemName.contains("\"")) {
                itemName = itemName.replace("'", "\'");
                StringBuilder xpath = new StringBuilder("//*[");
                String[] itemNameArray = itemName.split("\"");

                for (int i = 0; i < itemNameArray.length; i++) {
                    xpath.append(String.format("contains(text(), '%s')", itemNameArray[i]));
                    if (i < itemNameArray.length - 1) {
                        xpath.append(" and ");
                    }
                }

                xpath.append(String.format("]%s", brandXPathEnding));
                brandXPath = xpath.toString();
            } else {
                brandXPath = String.format("//*[contains(text(), '%s')]%s", itemName, brandXPathEnding);
            }
            WebElement brand = el.findElement(By.xpath(brandXPath));
            System.out.println(el.getText() + " " + brand.getText());
        });
    }`
And still there are too many conflicts, especially when it comes to an apostrophe inside double quotes (as in "Alice's Adventures in Wonderland"): it makes it almost impossible to provide a valid xpath for Selenium, because either way a String has quotes inside it:
<img width="886" alt="Screenshot 2023-01-10 at 23 24 23" src="https://user
-images.githubusercontent.com/122021944/211655710-d1118de7-2eca-4793-91dd-b503a530f1cf.png">
<img width="899" alt="Screenshot 2023-01-10 at 23 26 32" src="https://user-images.githubusercontent.com/122021944/211655739-f239a942-cf27-460d-8b30-f2ae268162a1.png">
 With the code above I can run tests for items both with double quotes and without them, but it creates too many exceptions in my particular case. I haven't commited this particular change and I wonder if there is a way out?
And yes, Carina is so much easier with all these cases :(

Reply:
I have found my way around the quotes and apostrophes, but still the xpath approach doesn't work as expected if the name of the item contains several languages. Updated method goes as:
`public void printItemData() {
        resultNameList.forEach(el -> {
            String brandXPath = null;
            String itemName = el.getText();
            StringBuilder xpath = new StringBuilder("//*[");
            String[] itemNameArray = itemName.split(" ");

            for (int i = 0; i < itemNameArray.length; i++) {
                itemNameArray[i] = itemNameArray[i].replaceAll("[\" ()]", "");
                xpath.append(String.format("contains(text(), \"%s\")", itemNameArray[i]));
                if (i < itemNameArray.length - 1) {
                    xpath.append(" and ");
                }
            }

            xpath.append(String.format("]%s", brandXPathEnding));
            brandXPath = xpath.toString();
            WebElement brand = el.findElement(By.xpath(brandXPath));
            System.out.println(el.getText() + " " + brand.getText());
        });
    }`

Reply:
Ok, got the point. Let's leave previous variant for now. 
But it's good that you find solution :)